grading/clean.cpp

```diff
@@ -308,7 +313,23 @@ bool isNumber(const std::string &str) {
       return false;
     }
   }
-  return true == atLeastOneDigit;
+  return atLeastOneDigit;
+}
+
+std::string isolateAlphanumAndNumberPunctuation(const std::string &str) {
+  std::string::const_iterator begin = str.begin();
+  while (!isalnum(*begin) && !(*begin == '.')) {
+    begin++;
+    if (begin == str.end())
+    {
+      return str;
```

Comment:
so you return the whole string if there is no number at all?
(this is not intuitive)